### PR TITLE
Some raid fixes

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #
Closes #
Closes #

# Changes proposed in this pull request:
- Guards fight when attacked even when sick or hungry
- Guards go to the hospital sooner when sick (directly when not attacked)
- Fleeing, doesnt wait each time for a hit, they do it a bit distance based
- Raid command doesnt let amazons slip through
- Raid command ignores config
- Cook stops giving food to visitors.

Needs testing later still

[ x ] Yes I tested this before submitting it.
[ ] I also did a multiplayer test.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/colonyEvents/raidEvents/HordeRaidEvent.java [hunk 1]
```diff
@@ -367,6 +364,7 @@ public void onStart()
         MessageUtils.format(RAID_EVENT_MESSAGE + horde.getMessageID(),
                         BlockPosUtil.calcDirection(colony.getCenter(), spawnPoint), colony.getName())
                 .with(ChatFormatting.DARK_RED).sendTo(colony).forManagers();
+        Log.getLogger().warn("Raiders coming from: " + spawnPoint.toShortString() + " towards colony: " + colony.getName());
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/colony/colonyEvents/raidEvents/HordeRaidEvent.java]: debug logging?

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/colonyEvents/raidEvents/HordeRaidEvent.java]: made it debug

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/CitizenAI.java [hunk 2]
```diff
@@ -133,7 +133,7 @@ private IState calculateNextState()
             }
 
             // Sick
-            if (citizen.getCitizenDiseaseHandler().isSick() && citizen.getCitizenJobHandler().getColonyJob().canAIBeInterrupted())
+            if (citizen.getCitizenDiseaseHandler().isSick() && citizen.getLastHurtMob() == null)
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/CitizenAI.java]: dont think this makes a lot of sense, AI interrupted is made for this purpose, now crafters are gonna stop mid craft, guards stop immediately after fighting even when there is more mobs around

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/CitizenAI.java]: This is only in the guard branch, so it doesnt affect crafters here.

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/CitizenAI.java]: The thing is, I believe AIInterrupt for eating is fine on the strictness, but I think, especially for guards, the strictness is too high for illness (Especially the super branch of the guard).

Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/CitizenAI.java]: what is wrong with AI interrupt here to begin with?

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/CitizenAI.java]: They run around for a while still while sick.

Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/CitizenAI.java]: well they have a combat timer for not immediately stopping fighting exactly for that reason they're supposed to keep fighting and not run away in the middle and die

**Review by someaddons — APPROVED** [review]

**Pull request merged**